src/AutoRest.CSharp/Common/Utilities/StringExtensions.cs

```diff
@@ -278,34 +280,56 @@ public static bool IsCSharpKeyword(string? name)
             }
         }
 
-
+        /// <summary>
+        /// Change a word to its plural form.
+        /// Notice that this function will treat this word as a whole word instead of only changing the last word if it contains multiple words. Please use <see cref="LastWordToPlural(string, bool)"/> instead.
+        /// </summary>
+        /// <param name="single"></param>
+        /// <param name="inputIsKnownToBeSingular"></param>
+        /// <returns></returns>
         public static string ToPlural(this string single, bool inputIsKnownToBeSingular = true)
         {
             return single.Pluralize(inputIsKnownToBeSingular);
         }
 
-        private static string LastWordToPlural(this string single, bool inputIsKnownToBeSingular = true)
+        public static string ResourceNameToPlural(this string resourceName)
```

Comment:
If this method is used only in mgmt, please move it into mgmt-specific type

Reply:
I move this function to the StringExtensions of Mgmt namespace